src/arch_ppc.py

```diff
@@ -0,0 +1,862 @@
+from typing import (
+    ClassVar,
+    Dict,
+    List,
+    Optional,
+    Set,
+    Tuple,
+)
+from .error import DecompFailure
+from .options import Target
+from .parse_instruction import (
+    Argument,
+    AsmAddressMode,
+    AsmGlobalSymbol,
+    AsmLiteral,
+    Instruction,
+    InstructionMeta,
+    JumpTarget,
+    Macro,
+    Register,
+)
+from .asm_pattern import (
+    AsmMatch,
+    AsmMatcher,
+    AsmPattern,
+    Replacement,
+    SimpleAsmPattern,
+    make_pattern,
+)
+from .translate import (
+    Abi,
+    AbiArgSlot,
+    Arch,
+    BinaryOp,
+    Cast,
+    CmpInstrMap,
+    CommentStmt,
+    ErrorExpr,
+    Expression,
+    ImplicitInstrMap,
+    InstrMap,
+    InstrSet,
+    PpcCmpInstrMap,
+    PairInstrMap,
+    SecondF64Half,
+    StmtInstrMap,
+    StoreInstrMap,
+    UnaryOp,
+    as_f32,
+    as_f64,
+    as_int64,
+    as_intish,
+    as_intptr,
+    as_ptr,
+    as_s32,
+    as_type,
+    as_u32,
+    fn_op,
+    fold_divmod,
+    fold_mul_chains,
+    handle_add,
+    handle_add_double,
+    handle_add_float,
+    handle_addi,
+    handle_addis,
+    handle_convert,
+    handle_load,
+    handle_loadx,
+    handle_or,
+    handle_rlwinm,
+    handle_sra,
+    load_upper,
+    make_store,
+    make_storex,
+    void_fn_op,
+)
+from .types import FunctionSignature, Type
+
+LENGTH_TWO: Set[str] = {
+    "neg",
+    "not",
+}
+
+LENGTH_THREE: Set[str] = {
+    "addi",
+    "ori",
+    "and",
+    "or",
+    "nor",
+    "xor",
+    "andi",
+    "xori",
+    "sllv",
+    "srlv",
+    "srav",
+}
```

Comment:
These seem wrong: missing a lot of instructions and `srlv` is MIPS. Do PPC assembly also have shorthands like MIPS where `ins $a, $b` expands to `ins $a, $a, $b` or can we rip this support out altogether?